component/prometheus/operator/common/crdmanager.go

```diff
@@ -39,15 +39,77 @@ import (
 // Generous timeout period for configuring all informers
 const informerSyncTimeout = 10 * time.Second
 
+type DiscoveryManager interface {
```

Comment:
Can you put these interfaces in a seperate file. I understand why you added them, and I do believe they are still useful, but this file is large enough as it is.